include/dev/TPS2HB50BQ1.hpp

```diff
@@ -0,0 +1,103 @@
+#ifndef TPS2HB50BQ1_HPP
+#define TPS2HB50BQ1_HPP
+
+#include <EVT/io/ADC.hpp>
+#include <EVT/io/CAN.hpp>
+#include <EVT/io/CANOpenMacros.hpp>
+#include <EVT/io/CANopen.hpp>
+#include <EVT/io/GPIO.hpp>
+
+namespace IO = EVT::core::IO;
+
+namespace LVSS {
+
+/**
+ * This class is used to control the TPS2HB50BQ1 power switch.
+ * The TPS2HB50BQ1 is a high side switch that can be used to control
+ * the power to up to two devices.
+ *
+ * The TPS2HB50BQ1 has a diagnostic mode that can be used to read the
+ * current, temperature, or fault status of the device.
+ */
+class TPS2HB50BQ1 {
+public:
+    TPS2HB50BQ1(IO::GPIO& en1, IO::GPIO& en2, IO::GPIO& senseOut, IO::GPIO& latch, IO::GPIO& diagEn,
+                IO::GPIO& diagSelect1, IO::GPIO& diagSelect2, IO::ADC& adc);
+
+    void setPowerSwitchStates(bool powerSwitchOneEnabled, bool powerSwitchTwoEnabled);
+
+    /**
+     * Get the fault status of the power switch
+     *
+     * @return The fault status of the power switch
+     */
+    uint32_t getFaultStatus();
+
+    /**
+     * Get the current of the power switch
+     *
+     * @return The current of the power switch
+     */
+    uint32_t getCurrent();
+
+    /**
+     * Get the temperature of the power switch
+     *
+     * @return The temperature of the power switch
+     */
+    uint32_t getTemp();
+
+    /**
+     * diagMode::OFF: Sets diagnostics pin to low, along with both diag select pins.
+     * diagMode::FAULT_STATUS: Get the fault status of the power switch
+     * diagMode::CURRENT: Get the current of the power switch
+     * diagMode::TEMP: Get the temperature of the power switch
+     */
+    enum DiagMode {
+        Off = 0x00,
```

Comment:
Enum values should be all caps
```suggestion
        OFF = 0x00,
```